examples/github-deploy.yml

```diff
@@ -0,0 +1,78 @@
+name: Build and upload to PyPI
+
+on: [push, pull_request]
+# alternatively, to publish when a (published) GitHub Release is created, use the following:
+# on:
+#   release:
+#     types:
+#       - published
+
+jobs:
+  build_wheels:
+    name: Build wheels on ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-18.04, windows-latest, macos-latest]
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - uses: actions/setup-python@v2
+        name: Install Python
+        with:
+          python-version: '3.7'
+
+      - name: Install cibuildwheel
+        run: |
+          python -m pip install cibuildwheel==1.4.2
+
+      - name: Install Visual C++ for Python 2.7
+        if: startsWith(matrix.os, 'windows')
+        run: |
+          choco install vcpython27 -f -y
+
+      - name: Build wheels
+        run: |
+          python -m cibuildwheel --output-dir wheelhouse
+
+      - uses: actions/upload-artifact@v2
+        with:
+          path: ./wheelhouse/*.whl
+
+  build_sdist:
+    name: Build source distribution
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - uses: actions/setup-python@v2
+        name: Install Python
+        with:
+          python-version: '3.7'
+
+      - name: Build sdist
+        run: python setup.py sdist
+
+      - uses: actions/upload-artifact@v2
+        with:
+          path: dist/*.tar.gz
+
+  upload_pypi:
+    needs: [build_wheels, build_sdist]
+    runs-on: ubuntu-latest
+    # upload to PyPI on every tag starting with 'v'
+    if: github.event_name == 'push' && startsWith(github.event.ref, 'refs/tags/v')
+    # alternatively, to publish when a GitHub Release is created, use the following rule:
+    # if: github.event_name == 'release' && github.event.action == 'published'
+    steps:
+      - uses: actions/download-artifact@v2
+        with:
+          name: artifact
```

Comment:
Since there's no way to download things using a wildcard, you could omit `name` arg and it'll download all the things. Then, it's pretty straightforward to mv dists into the `dist/` folder.

Reply:
If you add the name, it unpacks the folder, whereas if you don't, then it's in a nested folder. `upload-artifact@v2` combines automatically, so all the uploads go into the one artifact.

Reply:
Might be better to keep this like it is, then?

Reply:
Yes, it won't work if you change it unless you add a copy step. It's better to avoid a step, IMO. Though you could add a nested folder location in the next step, I suppose, but I think this is simpler.

Reply:
Oh, I guess when I implemented a similar thing previous time, it required the mv step because of v1. Will have to verify the new behaviour with v2.
OTOH I'm currently using different names in different workflows so I have a sync job for combining stuff anyway...

@henryiii do you know if v2 has any protection for race conditions when uploading with the same name?